src/dialog/docs/examples/primary-button-confirmation.example

```diff
@@ -0,0 +1,23 @@
+<DialogManager>
+  {({ isShown, isLoading, confirmLoading, show, hide }) => (
+    <Box>
+      <Dialog
+        isShown={isShown}
+        title="Dialog with Primary Confirmation"
+        onExited={hide}
+        primaryButton={{
+          children: isLoading ? 'Loading...' : 'Confirm Loading',
+          onClick: confirmLoading,
+          isLoading,
```

Comment:
I'm not sure how useful having `DialogManager` handle loading state is because in most real world cases the loading state will come down from the parent container.